### PR TITLE
A lot of typos fixed

### PR DESCRIPTION
**Pull request by tbw777**
BUILD SUCCESSFUL

### CODE REVIEW

**Review by vieiro — CHANGES_REQUESTED** [review]
Hi @tbw777 ,

We have already explained many times that you cannot change NetBeans public APIs at will, and that you have to coordinate with the rest of the team these changes in the mailing list before you submit the PR. 

Also please don't change unit test files and diff files blindly without understanding the implications of these. Otherwise you are introducing bugs.

Please review and update the PR accordingly to these guidelines.

Since we have already explained these guidelines many times in the PRs, please note that any future PRs that violate these guidelines will be immediately closed without any further notice.

**Comment by tbw777** [comment]
fixed

**Comment by vieiro** [comment]
Hi @tbw777 ,

Note that posting a PR for review, making reviewers review it, and then closing it is not very polite. 

If you continue doing this then no reviewers will spend their spare time in reviewing PRs that you then close at will, that's wasting time.

**Comment by tbw777** [comment]
@vieiro what commits do i need to revert?

File: enterprise/web.beans/test/unit/src/org/netbeans/modules/web/beans/model/InterceptorResolutionTest.java [hunk 1]
```diff
@@ -138,7 +138,7 @@ public void testInterceptorNonbindingMembers() throws IOException{
                 "@Target({METHOD, TYPE}) "+
                 "public @interface IBinding2  {" +
                 " String value() ;"+
-                " @Nonbinding String comment() defualt \"\" ; "+
+                " @Nonbinding String comment() default \"\" ; "+
```
Comment (vieiro) [inline enterprise/web.beans/test/unit/src/org/netbeans/modules/web/beans/model/InterceptorResolutionTest.java:141]: Are you changing a test here?

File: enterprise/web.jsf.navigation/src/org/netbeans/modules/web/jsf/navigation/graph/PageFlowSceneData.java [hunk 2]
```diff
@@ -45,7 +45,7 @@ public class PageFlowSceneData {
      * Because the actual Page object is not necessarily the same, it uses the display name
      * as the key.
      * @param scene PageFlowScene
-     * @param utilities PageFlowUtilites
+     * @param utilities PageFlowUtilities
```
Comment (vieiro) [inline enterprise/web.jsf.navigation/src/org/netbeans/modules/web/jsf/navigation/graph/PageFlowSceneData.java:48]: That's not a proper description of that "utilities" variable, is it?

File: groovy/groovy.editor/test/qa-functional/src/org/netbeans/test/groovy/cc/testCCMixed.java [hunk 3]
```diff
@@ -196,9 +196,9 @@ public void ImportPackage() {
         type(file, "\n public static int testInt(){\n return 0;");
         new EventTool().waitNoEvent(1000);
         file.save();
-        createGroovyFile(TEST_BASE_NAME + (name_iterator - 1), "Groovy Class", "EEE", "org.netbens.groovy");
+        createGroovyFile(TEST_BASE_NAME + (name_iterator - 1), "Groovy Class", "EEE", "org.netbeans.groovy");
```
Comment (vieiro) [inline groovy/groovy.editor/test/qa-functional/src/org/netbeans/test/groovy/cc/testCCMixed.java:199]: You're changing a test here. Did you test it afterwards?

File: ide/css.editor/src/org/netbeans/modules/css/editor/ui/CssRuleCreateActionDialog.java [hunk 4]
```diff
@@ -135,7 +135,7 @@ private void initComponents() {
         selectElementComboBox = new javax.swing.JComboBox();
         selectClassRadioButton = new javax.swing.JRadioButton();
         selectClassTextField = new javax.swing.JTextField();
-        selectElelmentIdRadioButton = new javax.swing.JRadioButton();
+        selectElementIdRadioButton = new javax.swing.JRadioButton();
```
Comment (vieiro) [inline ide/css.editor/src/org/netbeans/modules/css/editor/ui/CssRuleCreateActionDialog.java:138]: You're changing a variable name, did you change the corresponding .form file as well?

File: ide/css.editor/test/unit/data/testfiles/netbeans.css.formatted [hunk 5]
```diff
@@ -515,7 +515,7 @@ img.right {
 }
 /* -------------------------- */
 
-/* ----------- Docs & Suport spec styles -------------*/
+/* ----------- Docs & Support spec styles -------------*/
```
Comment (vieiro) [inline ide/css.editor/test/unit/data/testfiles/netbeans.css.formatted:518]: You're changing a test file, did you test the module afterwards?

File: ide/css.lib/src/org/netbeans/modules/css/lib/Css3.g [hunk 6]
```diff
@@ -2046,5 +2046,5 @@ LINE_COMMENT
     ;
 
 // -------------
-//  Illegal.    Any other character shoudl not be allowed.
+//  Illegal.    Any other character should not be allowed.
```
Comment (vieiro) [inline ide/css.lib/src/org/netbeans/modules/css/lib/Css3.g:2049]: You're changing a grammar file. Did you regenerate the grammar afterwards?

File: ide/css.visual/src/org/netbeans/modules/css/visual/SelectorsGroupEditor.java [hunk 7]
```diff
@@ -111,7 +111,7 @@ private void initComponents() {
         selectElementComboBox = new javax.swing.JComboBox();
         selectClassRadioButton = new javax.swing.JRadioButton();
         selectClassTextField = new javax.swing.JTextField();
-        selectElelmentIdRadioButton = new javax.swing.JRadioButton();
+        selectElementIdRadioButton = new javax.swing.JRadioButton();
```
Comment (vieiro) [inline ide/css.visual/src/org/netbeans/modules/css/visual/SelectorsGroupEditor.java:114]: You're changing an UI variable name, did you change the corresponding .form file as well?

File: ide/html.parser/external/htmlparser-01-fix-maven-build.patch [hunk 8]
```diff
@@ -45,7 +45,7 @@ diff --git a/src/nu/validator/htmlparser/dom/HtmlDocumentBuilder.java b/src/nu/v
 --- a/src/nu/validator/htmlparser/dom/HtmlDocumentBuilder.java
 +++ b/src/nu/validator/htmlparser/dom/HtmlDocumentBuilder.java
 @@ -377,26 +377,26 @@ public class HtmlDocumentBuilder extends
-     public void setTransitionHander(TransitionHandler handler) {
+     public void setTransitionHandler(TransitionHandler handler) {
```
Comment (vieiro) [inline ide/html.parser/external/htmlparser-01-fix-maven-build.patch:48]: You're changing a diff file here, are the offsets still correct? Or will this patch file be broken after your change?

File: java/i18n/src/org/netbeans/modules/i18n/wizard/AdditionalWizardPanel.java [hunk 9]
```diff
@@ -218,7 +218,7 @@ public static class Panel extends I18nWizardDescriptor.Panel {
         private transient AdditionalWizardPanel additionalPanel;
 
         Panel () {
-            emptyLabel = new JLabel(Util.getString("TXT_HasNoAdditonal"));//NOI18N
+            emptyLabel = new JLabel(Util.getString("TXT_HasNoAdditional"));//NOI18N
```
Comment (vieiro) [inline java/i18n/src/org/netbeans/modules/i18n/wizard/AdditionalWizardPanel.java:221]: Is this changed in the corresponding Bundle file?

File: ide/team.commons/src/org/netbeans/modules/team/spi/TeamAccessor.java [hunk 10]
```diff
@@ -35,7 +35,7 @@
 public abstract class TeamAccessor {
 
     public static final String PROP_LOGIN = "team.login.changed";               // NOI18N
-    public static final String PROP_PROJETCS_CHANGED = "team.projects.changed"; // NOI18N
+    public static final String PROP_PROJECTS_CHANGED = "team.projects.changed"; // NOI18N
```
Comment (vieiro) [inline ide/team.commons/src/org/netbeans/modules/team/spi/TeamAccessor.java:38]: You're changing a public NetBeans API/SPI without prior discussion.

File: enterprise/web.beans/test/unit/src/org/netbeans/modules/web/beans/model/InterceptorResolutionTest.java (hunk 1 above)
Comment (tbw777) [reply enterprise/web.beans/test/unit/src/org/netbeans/modules/web/beans/model/InterceptorResolutionTest.java:141]: image
over bindings are equal. Is this correct?

File: enterprise/web.jsf.navigation/src/org/netbeans/modules/web/jsf/navigation/graph/PageFlowSceneData.java (hunk 2 above)
Comment (tbw777) [reply enterprise/web.jsf.navigation/src/org/netbeans/modules/web/jsf/navigation/graph/PageFlowSceneData.java:48]: yes: Utilit i es

File: groovy/groovy.editor/test/qa-functional/src/org/netbeans/test/groovy/cc/testCCMixed.java (hunk 3 above)
Comment (tbw777) [reply groovy/groovy.editor/test/qa-functional/src/org/netbeans/test/groovy/cc/testCCMixed.java:199]: netbens fixed at 159fcdc8

**Comment by vieiro** [comment]
I added some quick reviews. A more thorough review will take much more time.

In this quick review, I've seen that:
- You correct some typos in internal variables, some documentation and elsewhere.
- You modify tests without really testing them (you say "BUILD SUCCESSFUL", but not "ALL TESTS OK").
- You modify diff files.
- You change public APIs/SPIs.
- This requires reviewer's time.

You have to understand that "Build successful" is not enough. 

You have to build, test and possibly run things before submitting the PRs, because we don't have time to spend reviewing hundreds of changes to files that break our tests and make our APIs backwards incompatible.

File: ide/css.editor/src/org/netbeans/modules/css/editor/ui/CssRuleCreateActionDialog.java (hunk 4 above)
Comment (tbw777) [reply ide/css.editor/src/org/netbeans/modules/css/editor/ui/CssRuleCreateActionDialog.java:138]: you are right here, fixed in new fdad92f1

File: ide/css.editor/test/unit/data/testfiles/netbeans.css.formatted (hunk 5 above)
Comment (tbw777) [reply ide/css.editor/test/unit/data/testfiles/netbeans.css.formatted:518]: dependings are changed at 6bf545f6

File: ide/css.lib/src/org/netbeans/modules/css/lib/Css3.g (hunk 6 above)
Comment (tbw777) [reply ide/css.lib/src/org/netbeans/modules/css/lib/Css3.g:2049]: This is comment line

File: ide/css.visual/src/org/netbeans/modules/css/visual/SelectorsGroupEditor.java (hunk 7 above)
Comment (tbw777) [reply ide/css.visual/src/org/netbeans/modules/css/visual/SelectorsGroupEditor.java:114]: 8000efbc fixed

File: ide/css.lib/src/org/netbeans/modules/css/lib/Css3.g (hunk 6 above)
Comment (matthiasblaesing) [reply ide/css.lib/src/org/netbeans/modules/css/lib/Css3.g:2049]: Do not change the grammar if there is no error to fix. I question fixing typos in non-user-facing code in general and in this case I plain object it.

File: ide/html.parser/external/htmlparser-01-fix-maven-build.patch (hunk 8 above)
Comment (tbw777) [reply ide/html.parser/external/htmlparser-01-fix-maven-build.patch:48]: This file
"Fix error in maven build when run on JDK8"
and therefore should be removed

Comment (tbw777) [reply ide/html.parser/external/htmlparser-01-fix-maven-build.patch]: file reverted in 537ad69d to remove in feature

File: java/i18n/src/org/netbeans/modules/i18n/wizard/AdditionalWizardPanel.java (hunk 9 above)
Comment (tbw777) [reply java/i18n/src/org/netbeans/modules/i18n/wizard/AdditionalWizardPanel.java:221]: yes

File: ide/team.commons/src/org/netbeans/modules/team/spi/TeamAccessor.java (hunk 10 above)
Comment (tbw777) [reply ide/team.commons/src/org/netbeans/modules/team/spi/TeamAccessor.java]: fixed in 0a41da3a
mark previos name as deprecated

**Comment by tbw777** [comment]
I agree that I need to run tests. I found how I can do it today. But I had a number of questions. It is enough to run the command: ant test?

File: ide/html.parser/external/htmlparser-01-fix-maven-build.patch (hunk 8 above)
Comment (matthiasblaesing) [reply ide/html.parser/external/htmlparser-01-fix-maven-build.patch]: > This file "Fix error in maven build when run on JDK8" and therefore should be removed

Don't guess. Especially not when you are fixing "typos".

Comment (tbw777) [reply ide/html.parser/external/htmlparser-01-fix-maven-build.patch]: ok, reverted

**Comment by vieiro** [comment]
> I agree that I need to run tests. I found how I can do it today. But I had a number of questions. Is it enough to run the command: ant test?

There are many kind of tests, ranging from unit tests to functional tests. Post a question in the mailing list to get some guidance, or consult the tutorials.

Note that the comments I've added are only those I found in a quick review.  You're expected to review **all of the rest of changes** in these few hundred files, and make sure that:

* you're not changing APIs/SPIs, 
* that all form you're changing can be opened in the IDE (this is, that Java code is in sync with the corresponding form and Bundle files).
* that all tests pass in the affected modules.
* that you can run the IDE (`ant tryme`) and that the forms that you've changed are working properly (this is, that no exception is thrown making the IDE or the form unusable).

Once you've done all these changes and tests then:

* squash the commits (`git rebase -i` may be of help) 
* and then force push to this same branch (`git push -f`). 
* Then ask for another review again.

We expect you're focused in this PR, making it pristine for our review, and that you won't open any other new PRs that do not match these quality criteria.

The goal of all this work is to have a new version of NetBeans that behaves **exactly the same** for end users (because users won't notice that you have renamed internal variables, because you have made sure that all tests work, and because you have made sure that forms don't break).

Thanks!

File: ide/team.commons/src/org/netbeans/modules/team/spi/TeamAccessor.java (hunk 10 above)
Comment (vieiro) [reply ide/team.commons/src/org/netbeans/modules/team/spi/TeamAccessor.java]: Marking things as deprecated is not enough when you're changing an API. To change a public API/SPI you have to:

* Ask in the mailing list if the change is acceptable or not.
* Ask in the mailing list for instructions on how to change a public API (marking things as "deprecated" is not good enough).

Thanks!

**Comment by tbw777** [comment]
To simplify the review process i replaced all changes by incomments only.
The rest will be commited in other PR.
Reson: most part of all changes are incomments only and its too hard to simultiniosly do analze different type of changes.

Checks status for current changes:
clean: ok
build : ok
ant rat: ok
ant verify-libs-and-licenses: ok
ant commit-validation: ok
ant tryme quick tests: no exceptions

**Comment by vieiro** [comment]
Hi @tbw777 ,

I think most NetBeans committers will understand NetBeans source code without a need to change any "typos" in comments. These can be changed in future commits when required. So correcting these typos is not useful nor for final users nor for developers. As explained in the mailing list, let's focus on tests, if there are any improvements in tests you can provide.

Also, quoting Matthias:

> Do not change the grammar if there is no error to fix. I question fixing typos in non-user-facing code in general and in this case I plain object it.

**Comment by tbw777** [comment]
Similar error on origin/master
What i need to do next?

File: php/php.editor/test/unit/data/testfiles/completion/lib/nb1855/nb1855.php.testCompleteAccessPrefixInClassConst_01.completion [hunk 11]
```diff
@@ -6,8 +6,8 @@ CLASS      ParentClass                     [PUBLIC]   nb1855.php
 CONSTANT   PRIVATE_CONSTANT "private cons  [PRIVATE]  Class1855
 CONSTANT   PROTECTED_CONSTANT "protected   [PROTECTE  Class1855
 CONSTANT   PROTECTED_PARENT_CONSTANT "pro  [PROTECTE  ParentClass
-CONSTANT   PUBLIC_CONSTANT "pubic constan  [PUBLIC]   Class1855
-CONSTANT   PUBLIC_PARENT_CONSTANT "pubic   [PUBLIC]   ParentClass
+CONSTANT   PUBLIC_CONSTANT "public consta  [PUBLIC]   Class1855
+CONSTANT   PUBLIC_PARENT_CONSTANT "public  [PUBLIC]   ParentClass
```
Comment (vieiro) [inline php/php.editor/test/unit/data/testfiles/completion/lib/nb1855/nb1855.php.testCompleteAccessPrefixInClassConst_01.completion:10]: So you're changing a ".completion" test file. 
Questions:
1) Can you explain how these completion test work?
2) Does this module pass an `ant test` after this change?

File: php/php.editor/test/unit/data/testfiles/completion/lib/nb1855/nb1855.php.testCompleteAccessPrefix_01.completion [hunk 12]
```diff
@@ -39,8 +39,8 @@ VARIABLE   ? publicTraitProperty           [PUBLIC]   Trait1855
 CONSTANT   PRIVATE_CONSTANT "private cons  [PRIVATE]  Class1855
 CONSTANT   PROTECTED_CONSTANT "protected   [PROTECTE  Class1855
 CONSTANT   PROTECTED_PARENT_CONSTANT "pro  [PROTECTE  ParentClass
-CONSTANT   PUBLIC_CONSTANT "pubic constan  [PUBLIC]   Class1855
-CONSTANT   PUBLIC_PARENT_CONSTANT "pubic   [PUBLIC]   ParentClass
+CONSTANT   PUBLIC_CONSTANT "public consta  [PUBLIC]   Class1855
+CONSTANT   PUBLIC_PARENT_CONSTANT "public  [PUBLIC]   ParentClass
```
Comment (vieiro) [inline php/php.editor/test/unit/data/testfiles/completion/lib/nb1855/nb1855.php.testCompleteAccessPrefix_01.completion:43]: Same two questions for this change.

File: php/php.editor/test/unit/data/testfiles/completion/lib/php54/traitsMultiUse.php.testTraitsMultiUse1.completion [hunk 13]
```diff
@@ -4,8 +4,8 @@ $this->|publicFirstField;
 METHOD     functionName()                  [PUBLIC]   WithMultiUses
 METHOD     privateFirstMethod()            [PRIVATE]  FirstTrait
 METHOD     privateSecondMethod()           [PRIVATE]  SecondTrait
-METHOD     protecedFirstMethod()           [PROTECTE  FirstTrait
-METHOD     protecedSecondMethod()          [PROTECTE  SecondTrait
+METHOD     protectedFirstMethod()          [PROTECTE  FirstTrait
+METHOD     protectedSecondMethod()         [PROTECTE  SecondTrait
```
Comment (vieiro) [inline php/php.editor/test/unit/data/testfiles/completion/lib/php54/traitsMultiUse.php.testTraitsMultiUse1.completion:8]: Here too.

**Comment by vieiro** [comment]
I still see little value in changing typos in the code. This is a > 500k LOC project, we really don't have time to review typos in half a million lines of code for developer-facing files. 

Are there any typos in user-facing files somewhere, perhaps?

**Comment by tbw777** [comment]
php tests reverted.
First step is to fix tests or we change plans?

File: php/php.editor/test/unit/data/testfiles/completion/lib/nb1855/nb1855.php.testCompleteAccessPrefixInClassConst_01.completion (hunk 11 above)
Comment (vieiro) [reply php/php.editor/test/unit/data/testfiles/completion/lib/nb1855/nb1855.php.testCompleteAccessPrefixInClassConst_01.completion]: What are the answers to these two questions?

Comment (tbw777) [reply php/php.editor/test/unit/data/testfiles/completion/lib/nb1855/nb1855.php.testCompleteAccessPrefixInClassConst_01.completion]: omg how long test is
1. no
2. too long execution to test. I run it with ant test from php/php.editor

**Comment by vieiro** [comment]
Hi @tbw777 ,

So we don't really want to correct typos in source code/comments that are not visible to the user. We prefer to focus the reviewers effort in things that have more impact on end users (solving bugs, adding new features, etc.). After all users don't bother what we name internal variables.

Modifying tests without understanding them is also problematic, as you could see for yourself tests take long time to run (specially PHP, which is tested thoroughly). Modifying test files to correct typos may alter the internal indexes we use in these files (specially in editor features) and may make tests to fail.

So what about closing this one and moving onto the next PR?

**Pull request closed without merging**